dashboard-frontend/src/helpers/arrays.ts

```diff
@@ -1 +1,28 @@
 export const lastValue = <T>(values: T[]): T => values[values.length - 1];
+
+interface Arr1 {
+  name: string;
+}
+
+interface Arr2 extends Arr1 {
+  data: string;
+}
+
+export const arrayMerge = (arr1: Arr1[], arr2: Arr2[] | any, key: string) => {
+  if (arr2.length > 0) {
+    return arr1.map(el => matcher(el, arr2, key));
+  }
+  return arr1;
+};
+
+const matcher = (el: any, arr2: [], key: string) => {
+  for (let i = 0; i < arr2.length; i++) {
+    if (el[key] === arr2[i][key]) {
+      return arr2[i];
+    }
+    if (!el[key] === arr2[i][key]) {
```

Comment:
What's the opposite of el[key]? I'm a bit confused by this one.

Reply:
Good point, not sure why I thought the second `if` statement was necessary...

Reply:
You're right! I realised this didn't make sense and noticed that it was actually exiting the loop at line 20 and running the condition again which gave me the desired result for the wrong reason at line 23. 

I've now refactored the if statement to reflect this.